### PR TITLE
Fix cooked steering review

### PR DESCRIPTION
**Pull request by erendrake**
I moved some of the code out into its own class. you will have to forgive me its getting late here so i wanted you to review and merge this if you dont find anything crazy.

### CODE REVIEW

File: src/kOS/Binding/SteeringManager.cs [hunk 1]
```diff
@@ -1391,7 +1361,7 @@ public void WriteCSVs()
             }));
         }
 
-        public void RemoveInstance(SharedObjects shared)
```
Comment (erendrake) [inline src/kOS/Binding/SteeringManager.cs]: we have a static and instance RemoveInstance that take different arguments? I hope i was successful in untangling them.

File: src/kOS/Binding/FlightControlManager.cs [hunk 2]
```diff
@@ -487,11 +488,21 @@ private void WheelSteer(FlightCtrlState c)
             public void Dispose()
             {
                 Enabled = false;
+                if (steeringManager != null)
+                {
+                    steeringManager.RemoveInstance(shared.Vessel);
```
Comment (hvacengi) [inline src/kOS/Binding/FlightControlManager.cs]: Should this use `SteeringManagerProvider.RemoveInstance` instead

File: src/kOS/Binding/FlightControlManager.cs [hunk 3]
```diff
@@ -487,11 +488,21 @@ private void WheelSteer(FlightCtrlState c)
...
+                    steeringManager = null;
+                }
             }
 
             public void UpdateFlightControl(Vessel vessel)
             {
                 control = GetControllerByVessel(vessel);
+                if (steeringManager != null)
+                {
+                    steeringManager = SteeringManager.SwapInstance(shared, steeringManager);
```
Comment (hvacengi) [inline src/kOS/Binding/FlightControlManager.cs:503]: Should I move `SwapInstance` into `SteeringManagerProvider`?

File: src/kOS/Binding/FlightControlManager.cs (hunk 2 above)
Comment (erendrake) [inline src/kOS/Binding/FlightControlManager.cs]: was that a question? I switched it

File: src/kOS/Binding/FlightControlManager.cs (hunk 3 above)
Comment (erendrake) [inline src/kOS/Binding/FlightControlManager.cs:503]: maybe, but i honestly decided that we just need to get this in and we can carve this thing up later.

**Comment by erendrake** [comment]
If you merge this into your PR branch and update it i say we merge this.

File: src/kOS/Binding/FlightControlManager.cs (hunk 2 above)
Comment (hvacengi) [inline src/kOS/Binding/FlightControlManager.cs]: Odd, I swear this was listed on the "files changed" page as still existing, but when I went to find it, I couldn't, and now it's shown as an outdated diff.  I guess it fixed itself.

**Comment by hvacengi** [comment]
Merged in.  I didn't actually have to do anything to manually merge it, even though github thought it couldn't be automatically merged.  But we should be good for now, until we find something else that breaks it.

**Comment by hvacengi** [comment]
Hold off on pulling, I was just running a test and I'm getting a stack overflow when disposing of the steering manager.  I'm looking into it now.

**Comment by hvacengi** [comment]
Turns out the problem was a circular reference between `SteeringManagerProvider.RemoveInstance` and `SteeringManager.Dispose` brought on by the change last night to use the provider's remove method last night.  The instance method `RemoveInstance` called dispose in a different order than the static method.  Because the instance method removed the key and instances from `AllInstances` before calling `Dispose`, it prevented the endless circular reference.  I'm testing an update now that does this same thing in the `SteeringManagerProvider` method.

**Pull request merged**